src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1612,6 +1612,25 @@ private void PerformPredicateAnalysisCore(IOperation operation, TAnalysisData ta
                         case OperationKind.BinaryPattern:
                             // These high level patterns should not be present in the lowered CFG: https://github.com/dotnet/roslyn/issues/47068
                             predicateValueKind = PredicateValueKind.Unknown;
+
+                            // We special case common null check to reduce false positives. But this implementation for BinaryPattern is very incomplete.
+                            if (FlowBranchConditionKind == ControlFlowConditionKind.WhenFalse)
```

Comment:
This is a nice optimization! Generic flow analysis handling for binary pattern operations is going to be non-trivial as there are sort of conditional branches with the `and` and `or` operators. I think this is a good starting point to handle the most important cases.